### PR TITLE
[local-authentication] Add support for `strong` and `weak` biometrics

### LINKED ISSUE #17804: Consistency issue between SecureStore and LocalAuthentication on Biometric strength
### Summary

SecureStore [rejects if `BIOMETRIC_STRONG` is not available](https://github.com/expo/expo/blob/21d4420641648e1fd9353121b31ec074506813cd/packages/expo-secure-store/android/src/main/java/expo/modules/securestore/AuthenticationHelper.kt#L128), with the following message:

> No hardware available for biometric authentication. Use expo-local-authentication to check if the device supports it.

However, [LocalAuthentication only checks if `BIOMETRIC_WEAK` or higher](https://github.com/expo/expo/blob/21d4420641648e1fd9353121b31ec074506813cd/packages/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt) authenticators are available.

Therefore, it is feasible that `LocalAuthentication` indicates that the device has authenticators, but `SecureStore` rejects on `requireAuthentication`.

Proposed solutions:

- Disallow weak authenticators entirely and search only for `BIOMETRIC_STRONG` in `LocalAuthentication`
  - I'm currently doing this locally with `patch-package`
- Or, allow the user to specify they only want `BIOMETRIC_STRONG` authenticators via a param to `supportedAuthenticationTypesAsync`, and document that only `BIOMETRIC_STRONG` is supported with `SecureStore` + `requireAuthentication`

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

_No response_

### Environment

This is not environment dependent.

### Reproducible demo

As this depends on hardware, this is not reproducible. Please view the links to the code in question.

### PR DESCRIPTION
**Pull request by behenate**
# Why

Android can use biometrics such as face scan that have lower security class than a fingerprint. We should allow users to decide if they want to allow such biometrics when authenticating, and also allow them to check if the device has enrolled strong or weak biometrics.

fixes https://github.com/expo/expo/issues/17804

# How

On Android added `biometricsSecurityLevel` field to `LocalAuthenticationOptions`, which can be either `strong` or `weak`. It defaults to `weak` to keep backwards compatibility. 

Values returned by `getEnrolledLevelAsync` have changed. We decided that the best action would be deprecating `SecurityLevel.BIOMETRIC` and introducing `SecurityLevel.BIOMETRIC_WEAK` and `SecurityLevel.BIOMETRIC_STRONG`.
 If someone uses `SecurityLevel.BIOMETRIC` it will be aliased into `BIOMETRIC_STRONG` on iOS (this should not change behaviour of the library as there are no weak biometrics on iOS) and `BIOMETRIC_WEAK` on Android. The android change might lead to changes in behaviour, so when someone uses `SecurityLevel.BIOMETRIC` a warning will be shown.

# Test Plan

Tested in BareExpo on Android 13 and iOS 16 devices.

### CODE REVIEW

**Review by lukmccall — APPROVED** [review]

File: packages/expo-local-authentication/src/LocalAuthentication.types.ts [hunk 1]
```diff
@@ -31,10 +33,45 @@ export enum SecurityLevel {
...
   /**
    * Indicates biometric authentication.
+   * @deprecated please use `BIOMETRIC_STRONG` or `BIOMETRIC_WEAK` instead.
+   * @hidden
+   */
+  BIOMETRIC = Platform.OS === 'android'
+    ? SecurityLevel.BIOMETRIC_WEAK
+    : SecurityLevel.BIOMETRIC_STRONG,
+  /**
+   * Indicates weak biometric authentication. For example 2D image-based face unlock.
```
Comment (amandeepmittal) [inline packages/expo-local-authentication/src/LocalAuthentication.types.ts:43]:
```suggestion
   * Indicates weak biometric authentication. For example, a 2D image-based face unlock.
```

File: packages/expo-local-authentication/src/LocalAuthentication.types.ts [hunk 2]
```diff
@@ -31,10 +33,45 @@ export enum SecurityLevel {
...
+   * > There are currently no weak biometric authentication options on iOS.
+   */
+  BIOMETRIC_WEAK = 2,
+  /**
+   * Indicates strong biometric authentication. For example a fingerprint scan or 3D face unlock.
```
Comment (amandeepmittal) [inline packages/expo-local-authentication/src/LocalAuthentication.types.ts:48]:
```suggestion
   * Indicates strong biometric authentication. For example, a fingerprint scan or 3D face unlock.
```

File: packages/expo-local-authentication/src/LocalAuthentication.types.ts [hunk 3]
```diff
@@ -61,6 +98,14 @@ export type LocalAuthenticationOptions = {
    * @platform android
    */
   requireConfirmation?: boolean;
+  /**
+   * Sets the security class of biometric authentication to allow.
+   * `strong` allows only Android Class 3 biometrics, for example a fingerprint or a 3D face scan.
+   * `weak` allows both Android Class 3 and Class 2 biometrics. Class 2 biometrics are less secure than Class 3, for example camera-based face unlock.
```
Comment (amandeepmittal) [inline packages/expo-local-authentication/src/LocalAuthentication.types.ts:104]:
```suggestion
   * Sets the security class of biometric authentication to allow.
   * `strong` allows only Android Class 3 biometrics. For example, a fingerprint or a 3D face scan.
   * `weak` allows both Android Class 3 and Class 2 biometrics. Class 2 biometrics are less secure than Class 3. For example, a camera-based face unlock.
```

**Review by amandeepmittal — APPROVED** [review]
Left some minor verbiage suggestions for consistency. Otherwise, docs updates are 👍

**Pull request merged**